src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution_ArgsToParameters.cs

```diff
@@ -164,6 +174,44 @@ public ImmutableArray<int> ToImmutableArray()
                 ArgumentAnalysisResult.NormalForm(argsToParameters.ToImmutableArray());
         }
 
+        private static int? CheckForBadNonTrailingNamedArgument(AnalyzedArguments arguments, ParameterMap argsToParameters, ImmutableArray<ParameterSymbol> parameters)
+        {
+            // Is there any named argument used out-of-position and followed by unnamed arguments?
+
+            // If the map is trivial then clearly not.
+            if (argsToParameters.IsTrivial)
+            {
+                return null;
+            }
+
+            // Find the first named argument which is used out-of-position or a params parameter
+            int foundPosition = -1;
+            int length = argsToParameters.Length;
+            for (int i = 0; i < length;  i++)
+            {
+                if (arguments.Name(i) != null && 
+                    (argsToParameters[i] != i || parameters[argsToParameters[i]].IsParams))
```

Comment:
I opened an issue https://github.com/dotnet/roslyn/issues/20249 to track problem with OverloadResolution.GetParameterType.